### PR TITLE
Doc updates #607

### LINKED ISSUE #607: Documentation for HTEX/EXEX
We need to add documentation for HTEX/EXEX including sample configs.

### PR DESCRIPTION
**Pull request by yadudoc**
Fixes #607, #424 

I wanted to post this early so that there's time to get more eyeballs on the doc.

### CODE REVIEW

File: docs/userguide/execution.rst [hunk 1]
```diff
@@ -178,8 +208,121 @@ tasks.
...
++---------------------+----------------------------+------------------------+
+| Cobalt based cluster| * `IPyParallelExecutor`    | `CobaltProvider`       |
+| or supercomputer    | * `HighThroughputExecutor` |                        |
+|                     | * `ExtremeScaleExecutor`   |                        |
++---------------------+----------------------------+------------------------+
+| GridEngine based    | * `IPyParallelExecutor`    | `GridEngineProvider`   |
+| cluster or grid     | * `HighThroughputExecutor` |                        |
++---------------------+----------------------------+------------------------+
+| Condor based        | * `IPyParallelExecutor`    | `CondorProvider`       |
+| cluster or grid     | * `HighThroughputExecutor` |                        |
```
[the remark this answers is not in the document]
Comment (yadudoc) [reply docs/userguide/execution.rst]: If you are in a grid environment, exex is not suitable.

File: docs/userguide/execution.rst [hunk 2]
```diff
@@ -178,8 +208,121 @@ tasks.
...
+|                     |                     |                    | * `IPyParallelExecutor`    |
+|                     |                     |                    | * `HighThroughputExecutor` |
++---------------------+---------------------+--------------------+----------------------------+
+| Nodes>1000          |  >minutes           |  0-1M              | * `ExtremeScaleExecutor`   |
++---------------------+---------------------+--------------------+----------------------------+
+
+3. ``If you are running on a cluster or supercomputer, will you request multiple nodes per block/job?``
+
++----------------------------------------------------------------------------+
+| If nodes_per_block = 1                                                     |
```
[the remark this answers is not in the document]
Comment (yadudoc) [reply docs/userguide/execution.rst]: fixed.

File: docs/userguide/execution.rst [hunk 3]
```diff
@@ -178,8 +208,121 @@ tasks.
 .. image:: parsl_parallelism.gif
 
 
+How-to Configure
```
Comment (annawoodard) [inline docs/userguide/execution.rst]: I think this whole table belongs in the configuration section.

File: docs/userguide/execution.rst [hunk 4]
```diff
@@ -178,8 +208,121 @@ tasks.
...
++---------------------+--------------------------+------------------------------------+
+| `SlurmProvider`     | Any                      | * `SrunLauncher`  if native slurm  |
+|                     |                          | * `AprunLauncher`, otherwise       |
++---------------------+--------------------------+------------------------------------+
+
+.. note:: If you are on a Cray system, you most likely need the `AprunLauncher` to launch workers unless you
+          are on a **native Slurm** system like :ref:`configuring_nersc_cori`
+
+
+4. ``Where will you run the workflow vs the tasks?``
```
[the remark this answers is not in the document]
Comment (yadudoc) [reply docs/userguide/execution.rst]: replaced `workflow` with `main parsl process`

File: docs/userguide/execution.rst [hunk 5]
```diff
@@ -178,8 +208,121 @@ tasks.
...
+|                     |                     |                    | * `HighThroughputExecutor` |
++---------------------+---------------------+--------------------+----------------------------+
+| 1<=Nodes<=1000      | <1s - minutes       |  0-1M              | * `ThreadPoolExecutor`     |
+|                     |                     |                    | * `IPyParallelExecutor`    |
+|                     |                     |                    | * `HighThroughputExecutor` |
++---------------------+---------------------+--------------------+----------------------------+
+| Nodes>1000          |  >minutes           |  0-1M              | * `ExtremeScaleExecutor`   |
++---------------------+---------------------+--------------------+----------------------------+
+
+3. ``If you are running on a cluster or supercomputer, will you request multiple nodes per block/job?``
```
[the remark this answers is not in the document]
Comment (yadudoc) [reply docs/userguide/execution.rst]: Fixed.

File: docs/userguide/execution.rst [hunk 6]
```diff
@@ -21,14 +32,33 @@ Executors represent a particular method via which tasks can be executed. As desc
...
+
+On many traditional batch systems, the user is expected to request a large number of nodes and launch tasks using a system such as `srun <https://slurm.schedmd.com/srun.html>`_ (for slurm), `aprun <https://cug.org/5-publications/proceedings_attendee_lists/2006CD/S06_Proceedings/pages/Authors/Karo-4C/Karo_alps_paper.pdf>`_ (for crays), `mpirun <https://www.open-mpi.org/doc/v2.0/man1/mpirun.1.php>`_ etc.
+Launchers are responsible for abstracting these different task-launch systems to start the appropriate number of workers across cores and nodes. Parsl currently supports the following set of launchers:
+
+1. `SrunLauncher`: Srun based launcher for Slurm based systems
+2. `AprunLauncher`: Aprun based launcher for Crays
+3. `SrunMPILauncher`: Launcher for launching MPI applications with Srun
+4. `GnuParallelLauncher`: Launcher using GNU parallel to launch workers across nodes and cores
+5. `MpiExecLauncher`: Uses Mpiexec to launch
+6. `SimpleLauncher`: The launcher default to a single worker launch.
```
[the remark this answers is not in the document]
Comment (yadudoc) [reply docs/userguide/execution.rst]: If you are setting `tasks_per_node` for the sake of the provider, and don't want that to interact with the launcher mechanism the SimpleLauncher would be useful.

**Pull request merged**